src/rabbit_amqqueue.erl

```diff
@@ -234,8 +234,14 @@ recover(VHost) ->
     %% for further processing in recover_durable_queues.
     {ok, OrderedRecoveryTerms} =
         BQ:start(VHost, [QName || #amqqueue{name = QName} <- Queues]),
-    {ok, _} = rabbit_amqqueue_sup_sup:start_for_vhost(VHost),
-    recover_durable_queues(lists:zip(Queues, OrderedRecoveryTerms)).
+    case rabbit_amqqueue_sup_sup:start_for_vhost(VHost) of
+        {ok, _} ->
+            recover_durable_queues(lists:zip(Queues, OrderedRecoveryTerms));
+        %% a failure
+        _       ->
+            rabbit_log:error("Failed to start queue supervisors for vhost '~s', skipping queue recovery", [VHost]),
+            []
```

Comment:
Should we raise the exception here? It's not safe to continue recovery process if the vhost supervisor is not found.